### PR TITLE
Add pillow requirement

### LINKED ISSUE #1533: Pillow required by wandb Image saving
There's an error with the latest prognostic_run image using wandb training where it's missing the package `pillow` for `wandb.Image`.  Need to update the requirements files.

### PR DESCRIPTION
**Pull request by frodre**
Resolves #1533

Requirement changes:

- Added `wandb[media]` requirement to fv3fit in order to upload wandb images
- [x] Ran make lock_deps/lock_pip following these instructions
- [x] Add PR review with license info for any additions to constraints.txt ([example](https://github.com/ai2cm/fv3net/pull/1218#pullrequestreview-663644359))

### CODE REVIEW

File: external/fv3fit/fv3fit/emulation/layers/architecture.py [hunk 1]
```diff
@@ -294,18 +294,6 @@ def call(self, hidden_output: tf.Tensor) -> Mapping[str, tf.Tensor]:
         }
 
 
-def _single_dense(rnn_out, dense_layer):
```
Comment (frodre) [inline external/fv3fit/fv3fit/emulation/layers/architecture.py:297]: I missed this unused function in the PR.

**Comment by oliverwm1** [comment]
FYI there are a couple checklist items in the PR template for updating dependencies. Could you do a license review for the new items in contraints.txt? For example: https://github.com/ai2cm/fv3net/pull/1218#pullrequestreview-663644359

File: docker/prognostic_run/requirements.txt [hunk 2]
```diff
@@ -84,10 +88,13 @@ packaging==20.4
 pandas==1.0.1
 partd==1.1.0
 pathtools==0.1.2
+pillow==8.4.0
 pint==0.15
 pkgconfig==1.5.1
+plotly==4.14.3
 pluggy==0.13.1
 pooch==1.1.1
+proglog==0.1.9
```
Comment (frodre) [inline docker/prognostic_run/requirements.txt:97]: MIT - copyright Edinburgh Genome Foundry

File: docker/prognostic_run/requirements.txt [hunk 3]
```diff
@@ -105,6 +112,7 @@ pytz==2020.1
 pyyaml==5.3
 requests-oauthlib==1.3.0
 requests==2.24.0
+retrying==1.3.3
```
Comment (frodre) [inline docker/prognostic_run/requirements.txt:115]: Apache Software License

File: docker/prognostic_run/requirements.txt [hunk 4]
```diff
@@ -113,6 +121,7 @@ shortuuid==1.0.1
 six==1.15.0
 smmap==4.0.0
 snowballstemmer==2.0.0
+soundfile==0.10.3.post1
```
Comment (frodre) [inline docker/prognostic_run/requirements.txt:124]: BSD

File: docker/prognostic_run/requirements.txt [hunk 5]
```diff
@@ -130,11 +139,13 @@ tensorflow==2.4.0
 termcolor==1.1.0
 toml==0.10.1
 toolz==0.10.0
+tornado==6.1
```
Comment (frodre) [inline docker/prognostic_run/requirements.txt:142]: Apache Software License

File: docker/prognostic_run/requirements.txt [hunk 6]
```diff
@@ -130,11 +139,13 @@ tensorflow==2.4.0
...
+tqdm==4.56.2
```
Comment (frodre) [inline docker/prognostic_run/requirements.txt:143]: MIT License; Mozilla Public License 2.0 (MPL 2.0)

File: docker/prognostic_run/requirements.txt [hunk 7]
```diff
@@ -130,11 +139,13 @@ tensorflow==2.4.0
...
 traitlets==4.3.3
 typeguard==2.10.0
 typing-extensions==3.7.4.3
 urllib3==1.25.10
-wandb==0.12.1
+wandb[media]==0.12.1
```
Comment (frodre) [inline docker/prognostic_run/requirements.txt:148]: MIT

**Review by frodre — COMMENTED** [review]
Add license for all updated packages

**Review by oliverwm1 — APPROVED** [review]
Thanks for doing the license review.

BTW, @nbren12 the dependency updating [instructions](https://vulcanclimatemodeling.com/docs/fv3net/dependency_management.html#dependency-management) say to run `make lock_deps` but in this case `make lock_pip` would have been sufficient given there was no change to `environment.yml`, right? That possibility is not mentioned in the docs. Anyway, maybe it's not a bad idea to update the conda lock files too.

**Comment by frodre** [comment]
@oliverwm1 @nbren12 

> but in this case make lock_pip would have been sufficient given there was no change to environment.yml, right? 

Maybe I should just revert the conda lockfile update to keep the update footprint smaller

**Comment by frodre** [comment]
Passed all the checks, so I'm going ahead with merge.

**Pull request merged**